sources/TileDB.CSharp/AggregateOperation.cs

```diff
@@ -0,0 +1,98 @@
+using TileDB.CSharp.Marshalling.SafeHandles;
+using TileDB.Interop;
+
+namespace TileDB.CSharp;
+
+/// <summary>
+/// Represents an aggregate operation.
+/// </summary>
+/// <remarks>
+/// This class abstracts the <c>tiledb_channel_operation_t</c> type of the TileDB C API.
+/// </remarks>
+public abstract class AggregateOperation
```

Comment:
I named this `AggregateOperation` instead of `ChannelOperation` or `QueryChannelOperation`. I believe it conveys its meaning better, and the first alternative is confusing (what is a channel?), and the second alternative is long.

This is a managed class that does not directly contain a `tiledb_channel_operation_t*`. Instead it encapsulates the logic to create it, which gets done when actually passed to the C API.